From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: stransky <stransky@redhat.com>
Date: Thu, 16 Jun 2022 06:22:33 +0000
Subject: [PATCH] [sailfishos][gecko] Bug 1758948 [FFmpeg] Use AVFrame::pts
 instead of AVFrame::pkt_pts on ffmpeg 4.x r=alwu

AVFrame::pkt_pts has been deprecated and gives us wrong values for AV1 VA-API. Let's use AVFrame::pts instead on ffmpeg 4.x
as well as we use on ffmpeg 5.0 where AVFrame::pkt_pts is removed.

Differential Revision: https://phabricator.services.mozilla.com/D149386
---
 dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp b/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp
index fdb33247bdb7..e925f21637a3 100644
--- a/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp
+++ b/dom/media/platforms/ffmpeg/FFmpegVideoDecoder.cpp
@@ -432,7 +432,7 @@ void FFmpegVideoDecoder<LIBAV_VER>::InitVAAPICodecContext() {
 #endif
 
 static int64_t GetFramePts(AVFrame* aFrame) {
-#if LIBAVCODEC_VERSION_MAJOR > 58
+#if LIBAVCODEC_VERSION_MAJOR > 57
   return aFrame->pts;
 #else
   return aFrame->pkt_pts;
